From: NIIBE Yutaka <gniibe@fsij.org>
Date: Wed, 12 Apr 2017 15:58:11 +0900
Subject: dirmngr: Fix possible null reference.

* dirmngr/dns.c (dns_error_t dns_trace_fput): Check NULL.

Signed-off-by: NIIBE Yutaka <gniibe@fsij.org>
(cherry picked from commit 7ae1857c90ab43ad9e31f0fb6dbd37f25cc37278)
---
 dirmngr/dns.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/dirmngr/dns.c b/dirmngr/dns.c
index 869e7ed..ebfd4c3 100644
--- a/dirmngr/dns.c
+++ b/dirmngr/dns.c
@@ -4594,8 +4594,9 @@ dns_error_t dns_trace_fput(const struct dns_trace_event *te, const void *data, s
 
 	if (fwrite(&tmp, 1, headsize, fp) < headsize)
 		return errno;
-	if (fwrite(data, 1, datasize, fp) < datasize)
-		return errno;
+	if (data)
+		if (fwrite(data, 1, datasize, fp) < datasize)
+			return errno;
 	if (fflush(fp))
 		return errno;
 
